src/test/basic.rs

```diff
@@ -45,16 +45,23 @@ fn struct_() -> Result<()> {
 				}
 			}
 
-			#[automatically_derived]
-			impl<T> ::core::cmp::Eq for Test<T> {
-				#[inline]
-				fn assert_receiver_is_total_eq(&self) {
-					struct __AssertEq<__T: ::core::cmp::Eq + ?::core::marker::Sized>(::core::marker::PhantomData<__T>);
+			const _: () = {
+				trait DeriveWhereAssertEq {
+					fn assert(&self);
+				}
 
-					// For some reason the comparison fails without the extra space at the end.
-					let _: __AssertEq<std::marker::PhantomData<T> >;
+				impl<T> DeriveWhereAssertEq for Test<T> {
+					fn assert(&self) {
+						struct __AssertEq<__T: ::core::cmp::Eq + ?::core::marker::Sized>(::core::marker::PhantomData<__T>);
```

Comment:
at some point we could consider creating our own types for stuff like that, though that'd require a non procmacro crate